controllers/chat/initChatFromProfileAsAnonymousV2.js

```diff
@@ -139,6 +116,29 @@ const initChatFromProfileAsAnonymousV2 = async (req, res) => {
         anon_user_info_emoji_name
       });
 
+    if (!createdChannel?.channel?.is_channel_blocked) {
+      chat = await channel.sendMessage({
+        user_id: req.userId,
+        text: message,
+        message_type: MESSAGE_TYPE.REPLY_PROMPT,
+        members,
+        message,
+        reply_data: {
+          user: {
+            id: member,
+            name: targetUserData.username,
+            image: targetUserData.profile_pic_path,
+            username: targetUserData.username
+          },
+          message: targetUserData.bio
+        },
+        anon_user_info_color_code,
+        anon_user_info_color_name,
+        anon_user_info_emoji_code,
+        anon_user_info_emoji_name
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The check for `createdChannel?.channel?.is_channel_blocked` is good, but there's a potential issue here. If `createdChannel?.channel` is `undefined`, this condition will pass and the code will attempt to send a message on an undefined channel, which would cause an error. To avoid this, you should also check if `createdChannel?.channel` exists before attempting to send a message.

```diff
-    if (!createdChannel?.channel?.is_channel_blocked) {
+    if (createdChannel?.channel && !createdChannel.channel.is_channel_blocked) {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->